### PR TITLE
Initial menu component

### PR DESCRIPTION
**Pull request by MartinForre**

### CODE REVIEW

File: src/components/general/Menu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        }
+
+        const items = props.sections[nextSectionIndex].items;
+
+        const itemIndex = items.findIndex(item => item.key === focusedItemKey);
+        let nextItemIndex = itemIndex;
+
+        if (nextItemIndex === -1) {
+            nextSectionIndex = direction === -1 ? props.sections.length - 1 : 0;
+            nextItemIndex = direction === -1 ? props.sections[nextSectionIndex].items.length - 1 : 0;
```
Comment (eskil0312) [inline src/components/general/Menu/index.tsx]: Could we use the items definition on line 33 here? 
`nextItemIndex = direction === -1 ? items.length - 1 : 0;`

File: src/components/general/Menu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        } else if (nextItemIndex === 0 && direction === -1) {
+            nextSectionIndex += (sectionIndex + props.sections.length - 1) % props.sections.length;
+        } else if (nextItemIndex === items.length - 1 && direction === 1) {
+            nextSectionIndex += (sectionIndex + props.sections.length + 1) % props.sections.length;
+        } else {
+            nextItemIndex += direction;
+        }
+
+        setFocusedSectionKey(props.sections[nextSectionIndex].key);
+        setFocusedItemKey(props.sections[nextSectionIndex].items[nextItemIndex].key);
```
Comment (eskil0312) [inline src/components/general/Menu/index.tsx]: items[nextItemIndex].key?

File: src/components/general/Menu/index.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+        if(props.sections[nextSectionIndex].items[nextItemIndex].isDisabled) {
```
Comment (eskil0312) [inline src/components/general/Menu/index.tsx]: items[nextItemIndex].isDisabled?

File: src/components/general/Menu/index.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+};
+
+export type MenuProps = {
+    sections: MenuSection[],
+    elevation?: Elevation,
+    onClick?: (item: MenuItemType) => void | Promise<void>,
+    keyboardNavigationRef?: HTMLElement | null,
+};
+
+const Menu: React.FC<MenuProps> = props => {
```
Comment (eskil0312) [inline src/components/general/Menu/index.tsx]: Would destructing the props here make the code more readable?

File: src/components/general/Menu/index.tsx (hunk 1 above)
Comment (MartinForre) [reply src/components/general/Menu/index.tsx]: Here the next section index has changed and items defined on line 33 no longer refers to the items in the "next section"

File: src/components/general/Menu/index.tsx (hunk 2 above)
Comment (MartinForre) [reply src/components/general/Menu/index.tsx]: The same issue as on line 40. Could possibly redefine items though 🤔

File: src/components/general/Menu/index.tsx (hunk 3 above)
Comment (MartinForre) [reply src/components/general/Menu/index.tsx]: The same issue as on line 40. Could possibly redefine items though 🤔

**Pull request merged**